manualCreditor.go

```diff
@@ -63,6 +62,10 @@ func (mc *manualCreditor) Drain(ctx context.Context) error {
 	// send drain, wait for responding flow frame
 	select {
 	case <-mc.drained:
+		// reset state here after the channel has been closed.
```

Comment:
I'm still not getting this. Is the problem that mc.drained might be nil before we reach line 65 because EndDrain happened so quickly?

Would it work the same if we just assigned the value of 'mc.drained' to a local 'drained' variable and then just <-'d on that? This would at least still make sure modifications to internals are within lock'd sections.

So, line 59ish:

```go
mc.drained = make(chan struct{})
drained := mc.drained
```

and then line 65ish:

```go
case <- drained:
```